pkgs/development/compilers/openjdk/openjfx/19.nix

```diff
@@ -0,0 +1,123 @@
+, libXxf86vm, glib, alsa-lib, ffmpeg_4, python3, ruby, icu68 }:
+
+let
+  major = "19";
+  update = "";
+  build = "+11";
+  repover = "${major}${update}${build}";
+  gradle_ = (gradle_7.override {
+    # note: gradle does not yet support running on 19
+    java = openjdk17_headless;
+  });
+
+  makePackage = args: stdenv.mkDerivation ({
+    version = "${major}${update}${build}";
+
+    src = fetchFromGitHub {
+      owner = "openjdk";
+      repo = "jfx";
+      rev = repover;
+      hash = "sha256-UXTaXtJ8py83V7IQK9wACIEWDAMRUaYNgH9e/Aeyuzc=";
+    };
+
+    patches = [
+      # 8295962: Reference to State in Task.java is ambiguous when building with JDK 19
+      (fetchpatch {
+        url = "https://github.com/openjdk/jfx/pull/933/commits/cfaee2a52350eff39dd4352484c892716076d3de.patch";
+        hash = "sha256-hzJMenhvtmHs/6BJj8GfaLp14myV8VCXCLLC8n32yEw=";
+      })
+      # ditto
+      (fetchpatch {
+        url = "https://github.com/openjdk/jfx/pull/933/commits/bd46ce12df0a93a56fe0d58d3653d08e58409b7f.patch";
+        hash = "sha256-o9908uw9vYvULmAh/lbfyHhgxz6jpgPq2fcAltWsYoU=";
+      })
+    ];
+
+    buildInputs = [ gtk2 gtk3 libXtst libXxf86vm glib alsa-lib ffmpeg_4 icu68 ];
+    nativeBuildInputs = [ gradle_ perl pkg-config cmake gperf python3 ruby ];
+
+    dontUseCmakeConfigure = true;
+
+    config = writeText "gradle.properties" (''
+      CONF = Release
+      JDK_HOME = ${openjdk19_headless.home}
+    '' + args.gradleProperties or "");
+
+    buildPhase = ''
+      runHook preBuild
+
+      export GRADLE_USER_HOME=$(mktemp -d)
+      ln -s $config gradle.properties
+      export NIX_CFLAGS_COMPILE="$(pkg-config --cflags glib-2.0) $NIX_CFLAGS_COMPILE"
+      gradle --no-daemon $gradleFlags sdk
+
+      runHook postBuild
+    '';
+  } // args);
+
+  # Fake build to pre-download deps into fixed-output derivation.
+  # We run nearly full build because I see no other way to download everything that's needed.
+  # Anyone who knows a better way?
+  deps = makePackage {
+    pname = "openjfx-deps";
+
+    # perl code mavenizes pathes (com.squareup.okio/okio/1.13.0/a9283170b7305c8d92d25aff02a6ab7e45d06cbe/okio-1.13.0.jar -> com/squareup/okio/okio/1.13.0/okio-1.13.0.jar)
+    installPhase = ''
+      find $GRADLE_USER_HOME -type f -regex '.*/modules.*\.\(jar\|pom\)' \
+        | perl -pe 's#(.*/([^/]+)/([^/]+)/([^/]+)/[0-9a-f]{30,40}/([^/\s]+))$# ($x = $2) =~ tr|\.|/|; "install -Dm444 $1 \$out/$x/$3/$4/$5" #e' \
+        | sh
+      rm -rf $out/tmp
+    '';
+
+    outputHashAlgo = "sha256";
+    outputHashMode = "recursive";
+    # suspiciously the same as for openjfx 17 ...
+    # could they really not have changed any of their dependencies?
+    # or did we miss changing another upstream hash when copy-pasting?
+    outputHash = "sha256-dV7/U5GpFxhI13smZ587C6cVE4FRNPY0zexZkYK4Yqo=";
+  };
+
+in makePackage {
+  pname = "openjfx-modular-sdk";
+
+  gradleProperties = ''
+    COMPILE_MEDIA = true
+    COMPILE_WEBKIT = false
```

Comment:
Is there any chance to enable this? JabRef uses webkit and currently uses the lib from maven with a hack.

https://github.com/NixOS/nixpkgs/blob/5a38d85664a26d71fa9a1eb404035c71a8d3d870/pkgs/applications/office/jabref/default.nix#L119

Reply:
I gave it a shot, but it's sort of tangled up with the mystery of the `sha256-dV7/...` hash a few lines above.  Even if I enable this for the `deps`, the output hash doesn't change!  So, the gradle build must be storing the things it fetches somewhere ad-hoc ... and so the main JavaFX build still tries & fails to fetch them.  :-/

linking explicitly to #162064 as well, as there's some discussion of this issue there too.  I'm afraid I don't have any more ideas on this front.  :(

Reply:
I thought now I understand the problem. As a workground, is that possible to use https://github.com/openjdk/jfx/blob/master/WEBKIT-MEDIA-STUBS.md#officially-released-libraries?

Reply:
Hm ... I _don't_ understand still!  :)  The page you linked to says these libraries are necessary for running tests.  I don't understand how the decision of whether or not to run the tests impacts whether `libjfxwebkit.so` is produced by the build or not, though.

This definitely feels like a discussion that's better had over in #162064, though.

Reply:
I mean maybe it's possible to use the webkit files from the maven central directly?

> Gradle has a task to automate downloading officially released libraries from MavenCentral.

Not sure if this can be used directly.

Reply:
> This definitely feels like a discussion that's better had over in https://github.com/NixOS/nixpkgs/issues/162064, though.

Yes. Maybe I can have a try after this is merged.